Saule/Serialization/ResourceSerializer.cs

```diff
@@ -90,10 +90,7 @@ private JToken SerializeMetadata()
             if (typeof(IEnumerable).IsAssignableFrom(valueType))
             {
                 isCollection = true;
-
-                // if our value is an enumerable, get the type of the items in the collection
-                var collectionType = valueType.GetInterfaces().FirstOrDefault(i => i.Name.Contains("IEnumerable") && i.IsGenericType);
-                valueType = collectionType?.GenericTypeArguments[0] ?? valueType;
+                valueType = valueType.TryGetCollectionType();
```

Comment:
If `GetGenericTypeParameterOfCollection` would return `null` if `valueType` was not a generic collection, this line could be

```csharp
valueType = valueType.GetGenericTypeParameterOfCollection() ?? valueType;
```

which would be more clear if you didn't know the context.